src/resource_server/views/api_v1/api.py

```diff
@@ -15,14 +15,38 @@ def cmd(endpoint):
     path_file = current_app.config["CMD_PATH_FILE"]
     base_url = current_app.config["SSH_KEYSIGN_BASE_URL"]
     user = current_user.claims[current_app.config["SSH_PRINCIPAL_CLAIM"]]
+    allowed_params = current_app.config["API_ALLOWED_QUERY_PARAMS"]
+
+
+
+    if request.method == 'POST':
+        params = request.json
+    elif request.method == 'GET':
+        params = request.args 
+    else: 
+        params = dict()
+
+    query_params = dict()
+    for key in params.keys():
+        if key not in allowed_params: 
+            continue
+        query_params[key] = params[key]
+    
+    request_params = create_request_param_to_config(query_params)
 
     # If the schema validator does raise any error then load the right endpoint values
-    command = load_template_values(validate_schema(path_file), endpoint)
+    command = load_template_values(validate_schema(path_file), endpoint, request_params)
 
     # Check the app env and get the mocking server port
     if current_app.config["TESTING"]:
-        command["port"] = request.json.get("ssh_port")
+        command["port"] = params.get("ssh_port")
+
+    headers = {
+        "User-Agent": request.headers.get("User-Agent"),
```

Comment:
User-Agent in this case should be the user agent calling ssh-service ...  not the original
same with Host .. Host should be the dns name of ssh-service, otherwise you will run into virtual hosting issues